### ISSUE TITLE
Why are `ISearchProvider` methods declared `async`?

### ISSUE THREAD

**Issue by AlbertHilb** [jupyterlab/jupyterlab#8264]
@jasongrout, @aschlaep

Hi all.

I'm working on the search and replace extension. I want to add an option to limit the scope of the search to the active cell (#2017) and to fix various other issues (#7881, #7898, #8152).

Looking at the code I encountered a design decision I don't understand.
I wish my doubts to be clarified before going on.

Why are the `ISearchProvider` interface methods declared `async`?
I can't find any place inside implementation of such methods, where asynchronous operations are executed.

Am I missing something?

**Comment by aschlaep** [comment]
Good question!  As I remember it, we decided to make the interface `async` in cases any future implementations of `ISearchProvider` needed to do something asynchronously.  Some examples we tossed around were searching a map or searching the filesystem (for whole-workspace search).  We thought it'd be best to just have the interface be `async` from the start, as it'd be really difficult to change it `async` once people started using it.
